### PR TITLE
Enabling FATES Satellite Phenology Mode

### PR DESCRIPTION
**Pull request by glemieux**
This pull request makes E3SM compatible with [FATES API 17.0.0](https://github.com/NGEET/fates/releases/tag/sci.1.47.0_api.17.0.0)
which enables the use of the [FATES satellite phenology mode](https://github.com/NGEET/fates/pull/703) (aka "sp mode").  

While FATES SP mode is operating, the state variables of `TLAI`, `TSAI`, `HTOP` and `WTCOL` are
manipulated (from the surface dataset files) in the native HLM PFT based index space. 
In this PR, we set up mapping from the FATES IFP space into the HLM P space in elmfates_interfaceMod.F90.  
Note that the `TLAI`, `TSAI`, and `HTOP` history outputs are updated with a logic check on
`use_fates_sp` to substitute the HLM variable output with the relevant 
fates sp mode version of these variables, which are identified with the suffix `_hist_patch`.  

This PR also introduces FATES no competition mode (`use_fates_nocomp`), 
which along with fixed biogeography mode (`use_fates_fixed_biogeog`), 
are necessary dependencies for fates sp mode.  
Fixed biogeography has been enabled in the namelist definition (previously it was set as a stub).  

The `SatellitePhenology` procedure has been slightly refactored to represent that it accepts different
filters depending on whether or not it is being called for FATES sp mode 
or ELM sp mode as the FATES sp mode implementation needs to have 
access to both the active and inactive soil filter.  

[nonBFB] FATES cases

### CODE REVIEW

File: components/elm/bld/namelist_files/namelist_defaults.xml [hunk 1]
```diff
@@ -122,10 +122,10 @@ attributes from the config_cache.xml file (with keys converted to upper-case).
 <!-- ================================================================== -->
 <!-- FATES default parameter file                                       -->
 <!-- ================================================================== -->
-<fates_paramfile >lnd/clm2/paramdata/fates_params_default_12pft_api16.c210527.nc</fates_paramfile>
+<fates_paramfile >lnd/clm2/paramdata/fates_params_api.16.1.0_12pft_c210630.nc</fates_paramfile>
```
Comment (rgknox) [inline components/elm/bld/namelist_files/namelist_defaults.xml:125]: FYI @glemiex, here is the new file: 
https://web.lcrc.anl.gov/public/e3sm/inputdata/lnd/clm2/paramdata/fates_params_12pft_api17.c211116.nc

File: components/elm/src/main/elmfates_interfaceMod.F90 [hunk 2]
```diff
@@ -1075,28 +1131,43 @@ subroutine wrap_update_hlmfates_dyn(this, nc, bounds_clump,      &
...
+               canopystate_inst%htop_hist_patch(p) = this%fates(nc)%bc_out(s)%htop_pa(ifp)
+             else
+               tlai(p) = this%fates(nc)%bc_out(s)%tlai_pa(ifp)
+               tsai(p) = this%fates(nc)%bc_out(s)%tsai_pa(ifp)
+               htop(p) = this%fates(nc)%bc_out(s)%htop_pa(ifp)
+             endif
+
+             if(use_fates_sp.and.abs(canopystate_inst%tlai_hist_patch(p) - &
+                this%fates(nc)%bc_out(s)%tlai_pa(ifp)).gt.1e-09)then
+               write(iulog,*) 'fates lai not like hlm lai',tlai(p),this%fates(nc)%bc_out(s)%tlai_pa(ifp),ifp
```
Comment (rgknox) [inline components/elm/src/main/elmfates_interfaceMod.F90:1168]: @glemieux should this be a fail?

File: components/elm/bld/namelist_files/namelist_defaults.xml (hunk 1 above)
Comment (glemieux) [reply components/elm/bld/namelist_files/namelist_defaults.xml:125]: Thanks @rgknox

File: components/elm/src/main/elmfates_interfaceMod.F90 (hunk 2 above)
Comment (glemieux) [reply components/elm/src/main/elmfates_interfaceMod.F90:1168]: Definitely.  Nice catch.

**Review by rgknox — APPROVED** [review]

**Comment by peterdschwartz** [comment]
It looks to me that this PR only needs a commit addressing the @rgknox review comments.  Is that correct?

**Comment by glemieux** [comment]
> It looks to me that this PR only needs a commit addressing the @rgknox review comments. Is that correct?

Yes and it also needs regression testing.  I should have this completed this week.

**Comment by rljacob** [comment]
@glemieux how did the testing go?

**Comment by glemieux** [comment]
@rljacob the `e3sm_land_developer` tests passed, although I'm seeing the MEMLEAK issue as reported here: https://github.com/E3SM-Project/E3SM/issues/4709.

I have a question out via email to Tesfa and Peter about how to handle topounits here: 
https://github.com/E3SM-Project/E3SM/pull/4669/files#diff-e1ab6a6e30962526c9e38886c09b164816ae6ff7331aef3d0d551e19c10d1cf8R652-R653

What I've got currently works, but I'm pretty sure this is incorrect and only working as I believe the datasets used for testing don't include topounits.  I need some guidance on how to collapse from a 3D array to a 2D array.

**Comment by glemieux** [comment]
I retested after committing 22cc2f7, to bring the fates science tag up to the last API 17 tag to make sure that https://github.com/NGEET/fates/commit/5de148e83c4cfb04576c8f16ed1a1e120d0184e7 was incorporated to deal with an IBM issue.  All tests pass B4B, although I am still seeing the MEMLEAK issue on the fates_rd testmod.

**Comment by peterdschwartz** [comment]
@glemieux I ran the e3sm_developer suite (after integrating to next) on chrysalis.  I didn't get any MEMLEAK issues, and only the fates tests DIFF.   Since the memleak issue is likely resolved by another PR, then is this ready to merge or are you waiting on the topounit clarification or anything else?

**Comment by glemieux** [comment]
Thanks for the update @peterdschwartz.  Tomorrow morning I have a meeting with @tktesfa2014 and @rgknox to discuss topounits, so I'd like to wait until then.  I think any update coming out from that will likely be fairly simple and quick to turn around.

**Comment by peterdschwartz** [comment]
Passed all the tests (just FATES diffs) after integration so I pushed it to next

**Comment by peterdschwartz** [comment]
Merged to master

**Pull request merged**